### PR TITLE
Use `ThreadFromGlobalPool` in `Poco::ThreadPool` for `server_pool` (to make global trace collector work for keeper and etc)

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Currently a draft, because this PR contains commits from https://github.com/ClickHouse/ClickHouse/pull/64896, so should be merged afterwards.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All NOT Required Checks
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [x]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with Aarch64
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
---
- [ ]  Do not test
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 7ea3345e0da818a6f9ea3127c9cffc54dbe9fd8a with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64922/7ea3345e0da818a6f9ea3127c9cffc54dbe9fd8a/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by antonio2368 — COMMENTED** [review]
LGTM

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -823,6 +818,11 @@ try
             total_memory_tracker.setSampleMaxAllocationSize(server_settings.total_memory_profiler_sample_max_allocation_size);
     }
 
+    auto server_pool = std::make_unique<Poco::ThreadPool>(3, server_settings.max_connections);
```
Comment (antonio2368) [inline programs/server/Server.cpp:821]: why was it moved?

Comment (kssenii) [reply programs/server/Server.cpp:821]: Bad merge with master, I did not notice it got moved.

File: programs/server/Server.cpp [hunk 2]
```diff
@@ -721,7 +721,7 @@ try
     CurrentMetrics::set(CurrentMetrics::Revision, ClickHouseRevision::getVersionRevision());
     CurrentMetrics::set(CurrentMetrics::VersionInteger, ClickHouseRevision::getVersionInteger());
 
-    Poco::ThreadPool server_pool(3, server_settings.max_connections);
+    auto server_pool = std::make_unique<Poco::ThreadPool>(3, server_settings.max_connections);
```
Comment (antonio2368) [inline programs/server/Server.cpp:724]: how does this affect anything?
maybe previous failures were related to accidentally moving the server_pool?

Comment (kssenii) [reply programs/server/Server.cpp:724]: There are two method of `Poco::ThreadPool`: `join()` and `release()`. The first if called on `servers_pool.joinAll()`, the second is called in destructor of `servers_pool`. The server hung on shutdown unless we call `release()` before `GlobalThreadPool::shutdown()`. May be second solution would be to change `join()` method to do what `release()` does, but I tried to leave it consistent with how it was.

**Comment by kssenii** [comment]
We decided to make it differently as this change is not user-friendly.

**Comment by kssenii** [comment]
Reimplemented in https://github.com/ClickHouse/ClickHouse/pull/65239.

**Pull request closed without merging**